### PR TITLE
vim-patch:9.0.0779: lsl and lm3 file extensions are not recognized

### PR DESCRIPTION
**Pull request by clason**
Problem:    lsl and lm3 file extensions are not recognized.
Solution:   Add *.lsl and *.lm3 patterns. (Doug Kearns, closes vim/vim#11384)
https://github.com/vim/vim/commit/4ac8e7948cb3e07bc4598ede8b274891d14dfa7c

### CODE REVIEW

**Comment by clason** [comment]
Feels good not having to port the Vim9script dist function to Vim1script, too :)

File: src/nvim/testdir/test_filetype.vim [hunk 1]
```diff
@@ -1962,4 +1962,35 @@ func Test_inc_file()
   filetype off
 endfunc
 
+func Test_lsl_file()
+  filetype on
+
+  call writefile(['looks like Linden Scripting Language'], 'Xfile.lsl', 'D')
```
Comment (seandewar) [inline src/nvim/testdir/test_filetype.vim:1968]: this requires defer

Comment (clason) [reply src/nvim/testdir/test_filetype.vim:1968]: D'oh! We _just_ talked about that, too...

Can we use it without the `D`?

File: runtime/lua/vim/filetype/detect.lua [hunk 2]
```diff
@@ -634,6 +634,19 @@ function M.lpc(bufnr)
   return 'c'
 end
 
+function M.lsl(bufnr)
+  if vim.g.filetype_lsl then
+    return vim.g.filetype_lsl
+  end
+
+  local line = nextnonblank(bufnr, 1)
+  if line:find('^%s*%') or line:find(':%s*trait%s*$') then
```
Comment (smjonas) [inline runtime/lua/vim/filetype/detect.lua:643]:
```suggestion
  if findany({ '^%s*%%', ':%s*trait%s*$' }) then
```

File: src/nvim/testdir/test_filetype.vim (hunk 1 above)
Comment (seandewar) [reply src/nvim/testdir/test_filetype.vim:1968]: yes but you'll need to `:call delete('Xfile.lsl')` manually.

that'll likely make this a partial port :thinking: (until there's consensus on what's happening in regards to porting defer)

File: src/nvim/testdir/test_filetype.vim [hunk 3]
```diff
@@ -1962,4 +1962,36 @@ func Test_inc_file()
   filetype off
 endfunc
 
+func Test_lsl_file()
+  filetype on
+
+  call writefile(['looks like Linden Scripting Language'], 'Xfile.lsl')
```
[the remark this answers is not in the document]
Comment (zeertzjq) [reply src/nvim/testdir/test_filetype.vim:1968]: Another to deal with this that I can think of is to just make `D` flag no-op. That avoids divergence in tests, but leaves files behind. Not sure what to do here.

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/testdir/test_filetype.vim:1968]: Wasn't there a plan to batch-convert the tests (from scratch?) via `vim9jit` when it's done?

[the remark this answers is not in the document]
Comment (zeertzjq) [reply src/nvim/testdir/test_filetype.vim:1968]: Does `vim9jit` also support legacy Vim script?

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/testdir/test_filetype.vim:1968]: Depends on the script.

But why is _this_ use of `'D'` different here? We haven't ported any of the other `'D'` related changes to this test, nor marked them somehow.

[the remark this answers is not in the document]
Comment (zeertzjq) [reply src/nvim/testdir/test_filetype.vim:1968]: > We haven't ported any of the other `'D'` related changes to this test, nor marked them somehow.

That's exactly why. This is the first time a non-skipped test that uses `D` flags has to be ported.

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/testdir/test_filetype.vim:1968]: I mean, the corresponding file in `vim/vim` uses `'D'` _everywhere_...

[the remark this answers is not in the document]
Comment (clason) [reply src/nvim/testdir/test_filetype.vim:1968]: To be clear: this file _already_ diverges from upstream in this regard even in the already existing tests. So marking _this specific change_ somehow would be misleading; if we do end up porting `defer` and the `D` flag, we need to re-sync the full file with upstream in any case.

[the remark this answers is not in the document]
Comment (zeertzjq) [reply src/nvim/testdir/test_filetype.vim:1968]: > if we do end up porting `defer` and the `D` flag, we need to re-sync the full file with upstream in any case.

Yes that's also an option for this file.

**Pull request merged**